ATSAMD RTC monotonic referral in documentation

kyp44 opened the pull request:
Per the discussion at the [2024-12-18 meeting](https://hackmd.io/c_mFUZL-Q2C6614MlrrxOg?view#Agenda-2024-12-18), this adds references in the documentation to the ATSAMD RTC monotonics now released in the [`atsamd-hal` crate](https://docs.rs/atsamd-hal/latest/atsamd_hal/rtc/rtic/index.html).

Specifically, a section about supported platforms was added to the `rtic-monotonics` README, as well as an ATSAMD section added to the main crate documentation.

I just took a stab at the language in the new sections so am certainly open to recommended changes.

Commit:
rtic-monotonics: Adds a section to the module documentation about ATSAMD chip monotonics provided in the `atsamd-hal` crate

Commit:
rtic-monotonics: Adds a section to README listing the supported microcontroller platforms.

AfoHT reviewed (dismissed):
Looks good!

I'll deal with the unrelated clippy fail and let's get this merged. Thank you 👍

AfoHT committed:
rtic-sync: Fix clippy complaints

error: the following explicit lifetimes could be elided: 'a
   --> rtic-sync/src/signal.rs:113:6
    |
113 | impl<'a, T: Copy> SignalReader<'a, T> {
    |      ^^                        ^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
    = note: `-D clippy::needless-lifetimes` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::needless_lifetimes)]`
help: elide the lifetimes
    |
113 - impl<'a, T: Copy> SignalReader<'a, T> {
113 + impl<T: Copy> SignalReader<'_, T> {

AfoHT committed:
xtask: Clippy fix warning

warning: elided lifetime has a name
   --> xtask/src/cargo_command.rs:427:14
    |
420 |     fn build_args<'i, T: Iterator<Item = &'i str>>(
    |                   -- lifetime `'i` declared here
...
427 |     ) -> Vec<&str> {
    |              ^ this elided lifetime gets resolved as `'i`
    |
    = note: `#[warn(elided_named_lifetimes)]` on by default
